File: src/medium/TopFrequentWords.swift
func topFrequentWords(_ words: [String], _ k: Int)-> [String] {
	var freq: [String: Int] = [:]
	for word in words {
		freq[word, default: 0] += 1
	}


	return freq.sorted{ (a, b) -> Bool in 
	if a.value != b.value {
		return a.value > b.value
	}
	return a.key < b.key
	
	}.prefix(k).map{ $0.key}
}

File: src/main/java/medium/TopFrequentWords.java
package medium;

import java.util.HashMap;
import java.util.List;

public class TopFrequentWords {
  public List<String> solve(String[] words, int k) {
    HashMap<String, Integer> freq = new HashMap<>();
    for (String word : words) {
      freq.put(word, freq.getOrDefault(word, 0) + 1);
    }
    return freq.entrySet().stream()
        .sorted(
            (a, b) -> {
              int f = a.getValue(), g = b.getValue();
              if (f != g) return g - f;
              return a.getKey().compareTo(b.getKey());
            })
        .map((entry) -> entry.getKey())
        .limit(k)
        .toList();
  }
}
